src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionSet.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.UnifiedSuggestions
+{
+    /// <summary>
+    /// Similar to SuggestedActionSet, but in a location that can be used
+    /// by both local Roslyn and LSP.
+    /// </summary>
+    internal class UnifiedSuggestedActionSet
+    {
+        public string? CategoryName { get; }
+
+        public IEnumerable<IUnifiedSuggestedAction> Actions { get; }
+
+        public object? Title { get; }
+
+        public UnifiedSuggestedActionSetPriority Priority { get; }
+
+        public TextSpan? ApplicableToSpan { get; }
+
+        public UnifiedSuggestedActionSet(
+            string? categoryName,
+            IEnumerable<IUnifiedSuggestedAction> actions,
+            object? title = null,
+            UnifiedSuggestedActionSetPriority priority = UnifiedSuggestedActionSetPriority.None,
+            TextSpan? applicableToSpan = null)
```

Comment:
for our own versions, consider not having optional parameters.  all clients on our end really should be explicit.